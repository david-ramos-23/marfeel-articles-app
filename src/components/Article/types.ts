export interface ArticleState {
  id: number;
  title: string;
  description: string;
  thumbnail?: string;
}

export interface StyledArticleProps {
  onError: any;
}
