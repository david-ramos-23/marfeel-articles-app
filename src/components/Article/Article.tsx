import React from 'react';
import { ArticleState } from './types';
import {
  StyledArticleContainer,
  StyledArticleImageContainer,
  StyledArticleThumbnail,
  StyledArticleInfoContainer,
  StyledArticleTitle,
  StyledArticleDescription,
} from './styled';

export const Article: React.FunctionComponent<ArticleState> = ({
  title,
  description,
  thumbnail,
}) => {
  const [urlError, setUrlError] = React.useState(null);
  const hasThumbnail = thumbnail && !urlError;

  return (
    <StyledArticleContainer>
      {hasThumbnail && (
        <StyledArticleImageContainer>
          <StyledArticleThumbnail
            src={thumbnail}
            onError={setUrlError}
            alt="thumbnail"
          />
        </StyledArticleImageContainer>
      )}

      <StyledArticleInfoContainer>
        <StyledArticleTitle>{title}</StyledArticleTitle>
        <StyledArticleDescription>{description}</StyledArticleDescription>
      </StyledArticleInfoContainer>
    </StyledArticleContainer>
  );
};

export default Article;
