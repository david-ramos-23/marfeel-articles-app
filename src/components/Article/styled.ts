import styled from 'styled-components';
import { StyledArticleProps } from './types';

export const StyledArticleContainer = styled.div`
  display: flex;
  flex-direction: row;
  padding: 20px;
`;

export const StyledArticleImageContainer = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  margin-right: ${(props) => props.theme.sizes.size2};
`;

export const StyledArticleInfoContainer = styled.div`
  flex: 2;
  display: flex;
  flex-direction: column;
`;

export const StyledArticleThumbnail = styled.img<StyledArticleProps>`
  max-width: 100%;
  height: auto;
`;

export const StyledArticleTitle = styled.h1`
  font-family: ${(props) => props.theme.articles.fontFamily};
  font-size: ${(props) => props.theme.sizes.size2};
  text-transform: capitalize;
  padding-bottom: ${(props) => props.theme.sizes.size1};
`;

export const StyledArticleDescription = styled.p`
  font-family: ${(props) => props.theme.articles.fontFamily};
`;
