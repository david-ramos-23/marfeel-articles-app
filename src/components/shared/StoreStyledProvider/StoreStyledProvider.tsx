import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import configureStore from '../../../store';
import { defaultTheme } from '../../../store/theme/theme';
import { SectionProvider } from '../../../hooks/useSection';

export const StoreStyledProvider: React.FunctionComponent = ({ children }) => (
  <Provider store={configureStore()}>
    <ThemeProvider theme={defaultTheme}>
      <SectionProvider>
        <>{children}</>
      </SectionProvider>
    </ThemeProvider>
  </Provider>
);
