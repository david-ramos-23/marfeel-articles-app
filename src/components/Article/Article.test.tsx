import React from 'react';
import { shallow } from 'enzyme';
import Article from './Article';
import {
  StyledArticleTitle,
  StyledArticleDescription,
  StyledArticleThumbnail,
} from './styled';
import renderer from 'react-test-renderer';
import { StoreStyledProvider } from '../shared/StoreStyledProvider';

describe('<Article />', () => {
  const props = {
    id: 0,
    title: 'Title',
    description: 'Description',
  };

  it('renders correctly', () => {
    const wrapper = renderer
      .create(
        <StoreStyledProvider>
          <Article {...props} />
        </StoreStyledProvider>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders title and description', () => {
    const wrapper = shallow(<Article {...props} />);
    expect(wrapper.find(StyledArticleTitle).text()).toEqual(props.title);
    expect(wrapper.find(StyledArticleDescription).text()).toEqual(
      props.description
    );
  });

  it('renders without an image', () => {
    const wrapper = shallow(<Article {...props} />);
    expect(wrapper.find(StyledArticleThumbnail)).toHaveLength(0);
  });

  it('renders with an image', () => {
    const wrapper = shallow(<Article {...props} thumbnail="some.cool.png" />);
    expect(wrapper.find(StyledArticleThumbnail)).toHaveLength(1);
  });
});
