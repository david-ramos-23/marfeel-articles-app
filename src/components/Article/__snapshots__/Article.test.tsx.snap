// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> renders correctly 1`] = `
<div
  className="sc-AxjAm hMzSbb"
>
  <div
    className="sc-AxiKw dMqpCe"
  >
    <h1
      className="sc-AxhUy bXZbHk"
    >
      Title
    </h1>
    <p
      className="sc-AxgMl fNmGFA"
    >
      Description
    </p>
  </div>
</div>
`;
